test/e2e/network_policy.go

```diff
@@ -0,0 +1,440 @@
+	// default.  Create another and store both separately for clarity.
+	nsA := f.Namespace
+	nsB, err := f.CreateNamespace(f.BaseName+"-b", map[string]string{
+		"e2e-framework": f.BaseName + "-b",
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	// Turn on isolation
+	setNetworkIsolationAnnotations(f, nsA, true)
+	setNetworkIsolationAnnotations(f, nsB, true)
+
+	// Get the available nodes.
+	nodes, err := framework.GetReadyNodes(f)
+	framework.ExpectNoError(err)
+
+	if len(nodes.Items) == 1 {
+		// in general, the test requires two nodes. But for local development, often a one node cluster
+		// is created, for simplicity and speed. (see issue #10012). We permit one-node test
+		// only in some cases
+		if !framework.ProviderIs("local") {
+			framework.Failf(fmt.Sprintf("The test requires two Ready nodes on %s, but found just one.", framework.TestContext.Provider))
+		}
+		framework.Logf("Only one ready node is detected. The test has limited scope in such setting. " +
+			"Rerun it with at least two nodes to get complete coverage.")
+	}
+
+	// Create service A and B.  These are really just used
+	// for pod discovery by the net monitor containers.
+	serviceA := createService(f, nsA, serviceAName)
+	serviceB := createService(f, nsB, serviceBName)
+
+	// Clean up services
+	defer func() {
+		By("Cleaning up the service A")
+		if err = f.Client.Services(nsA.Name).Delete(serviceA.Name); err != nil {
+			framework.Failf("unable to delete svc %v: %v", serviceA.Name, err)
+		}
+	}()
+	defer func() {
+		By("Cleaning up the service B")
+		if err = f.Client.Services(nsB.Name).Delete(serviceB.Name); err != nil {
+			framework.Failf("unable to delete svc %v: %v", serviceB.Name, err)
+		}
+	}()
+
+	By("Creating a webserver (pending) pod on each node")
+	podAName, podBNames := launchNetMonitorPods(f, nsA, nsB, nodes)
+
+	// Deferred clean up of the pods.
+	defer func() {
+		By("Cleaning up the webserver pods")
+		if err = f.Client.Pods(nsA.Name).Delete(podAName, nil); err != nil {
+			framework.Logf("Failed to delete pod %s: %v", podAName, err)
+		}
+		for _, podName := range podBNames {
+			if err = f.Client.Pods(nsB.Name).Delete(podName, nil); err != nil {
+				framework.Logf("Failed to delete pod %s: %v", podName, err)
+			}
+		}
+	}()
+
+	// Wait for all pods to be running.
+	By(fmt.Sprintf("Waiting for pod %q to be running", podAName))
+	err = framework.WaitForPodRunningInNamespace(f.Client, podAName, nsA.Name)
+	Expect(err).NotTo(HaveOccurred())
+	for _, podName := range podBNames {
+		By(fmt.Sprintf("Waiting for pod %q to be running", podName))
+		err = framework.WaitForPodRunningInNamespace(f.Client, podName, nsB.Name)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// Open up port 8080 so that we can access the network monitor port
+	By("Checking full isolation")
+	addNetworkPolicyOpenPort(f, nsA, "ui1", "8080", "TCP")
+	addNetworkPolicyOpenPort(f, nsB, "ui2", "8080", "TCP")
+
+	// We expect full isolation between namespaces.  Monitor, wait for a little and
```

Comment:
"We expect full isolation between all pods in all namespaces"

We don't expect the pods in namespace B to be able to communicate (even though they are in the same namespace)